sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSinkLog.scala

```diff
@@ -96,6 +97,19 @@ class FileStreamSinkLog(
   require(defaultCompactInterval > 0,
     s"Please set ${SQLConf.FILE_SINK_LOG_COMPACT_INTERVAL.key} (was $defaultCompactInterval) " +
       "to a positive value.")
+
+  private val ttlMs = outputTimeToLiveMs.getOrElse(Long.MaxValue)
+
+  override def shouldRetain(log: SinkFileStatus): Boolean = {
+    val curTime = System.currentTimeMillis()
```

Comment:
It would be great to avoid calling `System.currentTimeMillis()` if the option is not set, considering we need to call this method once (a JNI call) for each log entry.

Reply:
Probably we could change the method signature a bit to provide "context" which would be same for the same compact batch. We changed the method `shouldRetain` in SPARK-30462 which is not yet released (3.1.0), hence making change shouldn't make backward compatibility. (We decided to break the interface, but we break only once for these changes.)

Reply:
Ah also `CompactibleFileStreamLog` is not a public API (its package is `org.apache.spark.sql.execution.streaming`), so it shouldn't matter much.